From 0be62ffe5de0eebde218d5a1146ca7a9666a55ba Mon Sep 17 00:00:00 2001
From: Shashi Shekar B.S <shashi.shekar@ti.com>
Date: Fri, 15 Jan 2010 11:48:22 +0530
Subject: [PATCH] Changes needed for multi-framebuffer devices.

Multi-frambuffer devices do not sit on ISA or PCI bus, so this
hack is necessary to detect/ init multi-framebuffer devices on OMAP4.
---
 hw/xfree86/fbdevhw/fbdevhw.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/hw/xfree86/fbdevhw/fbdevhw.c b/hw/xfree86/fbdevhw/fbdevhw.c
index 5269277..0959723 100644
--- a/hw/xfree86/fbdevhw/fbdevhw.c
+++ b/hw/xfree86/fbdevhw/fbdevhw.c
@@ -457,6 +457,11 @@ fbdevHWInit(ScrnInfoPtr pScrn, struct pci_device * pPci, char *device)
 	fbdevHWGetRec(pScrn);
 	fPtr = FBDEVHWPTR(pScrn);
 
+	if (pScrn->scrnIndex == 0)
+	 device = "/dev/fb0";
+	else 
+	 device = "/dev/fb1";
+
 	/* open device */
 	if (pPci)
 		fPtr->fd = fbdev_open_pci(pPci,NULL);
-- 
1.5.6.3

